plan/plan.go

```diff
@@ -65,24 +65,16 @@ func (p *Plan) Calculate() *Plan {
 			continue
 		}
 
-		targetChanged := targetChanged(desired, current)
 		shouldUpdateTTL := shouldUpdateTTL(desired, current)
 
-		if !targetChanged && !shouldUpdateTTL {
+		if !shouldUpdateTTL {
```

Comment:
This condition skipped the record in case nothing was changed (target nor TTL). With this it skips the record whenever TTL was changed (but target could have) which seems wrong to me. Can you double-check?

Reply:
`desired` and `current` are guaranteed to have the same target (see the recordExists() modification below), so targetChanged would always be false. That's why I removed it. As I said, with this PR you'd have one `Endpoint` per DNSName and Target rather than just per DNSName, so if the target changes, the corresponding endpoint will be removed and a new one with the same name and the new target created. So it treats target changes just like it always treated name changes -- the old Endpoint will be removed and a new one created. The TTL would be the only remaining thing in an Endpoint that can actually "change".

That's really the crux of this PR -- don't turn `Endpoint.Target` into an array, instead have one Endpoint per name and target. I did this because I figured it would require fewer (possibly no) changes to the providers, and I wanted to get the CF provider up and running with multiple targets quickly (I initially thought you'd basically get away with just changing plan.recordExists(), then saw that I should also modify the ownerId tracking in TXTRegistry). I guess we'd have to discuss whether this is the best way forward or not, since this is obviously incompatible with the other multitarget PRs that turn `Endpoint.Target` into an array.